dapr/proto/runtime/v1/dapr.proto

```diff
@@ -536,3 +542,56 @@ message UnsubscribeConfigurationResponse {
   string message = 2;
 }
 
+message TryLockRequest {
+  // Required. The lock store name,e.g. `redis`.
+  string store_name = 1;
+
+  // Required. resource_id is the lock key. e.g. `order_id_111`
+  // It stands for "which resource I want to protect"
+  string resource_id = 2;
+
+  // Required. lock_owner indicate the identifier of lock owner.
+  // You can generate a uuid as lock_owner.For example,in golang:
+  //
+  // req.LockOwner = uuid.New().String()
+  //
+  // This field is per request,not per process,so it is different for each request,
+  // which aims to prevent multi-thread in the same process trying the same lock concurrently.
+  //
+  // The reason why we don't make it automatically generated is:
+  // 1. If it is automatically generated,there must be a 'my_lock_owner_id' field in the response.
+  // This name is so weird that we think it is inappropriate to put it into the api spec
+  // 2. If we change the field 'my_lock_owner_id' in the response to 'lock_owner',which means the current lock owner of this lock,
+  // we find that in some lock services users can't get the current lock owner.Actually users don't need it at all.
+  // 3. When reentrant lock is needed,the existing lock_owner is required to identify client and check "whether this client can reenter this lock".
+  // So this field in the request shouldn't be removed.
+  string lock_owner = 3;
```

Comment:
What happens if two calls from the same app use the same `lock_owner` on the same `resource_id`?

Reply:
case 1.  two `Trylock` calls from the same app use the same lock_owner on the same resource_id
It depends on whether the lock component is reentrant. Currently we didn't force the lock component to be reentrant. But we can add some restriction in the next iteration.

case 2. two callers (maybe in two threads) invoke `Unlock` from the same app use the same `lock_owner` on the same `resource_id`
One caller might unexpectedly release the other's lock

Reply:
So what will happen in the current iteration for case 1?

Reply:
@yaron2 The second TryLock will fail